v0/destinations/google_ads_classic/transform.js

```diff
@@ -0,0 +1,156 @@
+const get = require("get-value");
+const {
+  trackConfig,
+  baseEndpoint,
+  reservedEventNames,
+  mappedEventTypes
+} = require("./config");
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  removeUndefinedValues
+} = require("../../util");
+const { EventType } = require("../../../constants");
+
+/**
+ *
+ * @param {Rudder Payload} message
+ * To set app_event_type
+ * The name of the app event that occurred. This field is an enumeration, and will only accept valued predefined by google ads.
+ * https://developers.google.com/app-conversion-tracking/api/request-response-specs#conversion_tracking_request
+ *
+ */
+const setEventType = message => {
+  const eventName = get(message, "event");
+  let eventType;
+  let eventValue;
+  Object.keys(mappedEventTypes).forEach(mappedEventType => {
+    if (mappedEventType === eventName) {
+      eventType = mappedEventType;
+      eventValue = mappedEventTypes[mappedEventType];
+    }
+  });
+  switch (eventName.toLowerCase()) {
+    case "order completed":
+      // Use in_app_purchase for purchases made through the native app store; use ecommerce_purchase for all other purchases.
+      if (
+        get(message, "properties.affiliation") === "App Store" ||
+        get(message, "properties.affiliation") === "Google Store"
+      ) {
+        return "in_app_purchase";
+      }
+      return "ecommerce_purchase";
+    case eventType:
+      return eventValue;
+    default:
+      return "custom";
+  }
+};
```

Comment:
```suggestion
const setEventType = message => {
  const evName = get(message, "event").toLowerCase();
  const mappedEvent = mappedEventTypes[evName];
  if (mappedEvent) {
    return mappedEvent;
  }

  if (evName === "order completed") {
      if (
        get(message, "properties.affiliation") === "App Store" ||
        get(message, "properties.affiliation") === "Google Store"
      ) {
        return "in_app_purchase";
      }
      return "ecommerce_purchase";
  }

  return "custom";
};
```
I think removing the loop will be better here.

Reply:
yes